File: backendfastapi/agents/agent_factory.py
from __future__ import annotations

from typing import List

from langchain.agents import create_tool_calling_agent, AgentExecutor
from langchain.chat_models.base import BaseChatModel
from langchain_core.prompts import ChatPromptTemplate
from langchain_core.tools import Tool


def create_agent(
    llm: BaseChatModel,
    tools: List[Tool],
    prompt: ChatPromptTemplate,
) -> AgentExecutor:
    """Construct and return a *bare* LangChain AgentExecutor.

    The caller is responsible for wrapping the executor with memory,
    output parsing, etc.  This function focuses solely on composing the
    agent graph (LLM + tools + prompt).
    """
    agent = create_tool_calling_agent(llm, tools, prompt)
    return AgentExecutor(agent=agent, tools=tools, verbose=True)

File: backend/src/agents/agentFactory.ts
import { createToolCallingAgent, AgentExecutor } from 'langchain/agents';
import { BaseChatModel } from '@langchain/core/language_models/chat_models';
import { ChatPromptTemplate } from '@langchain/core/prompts';
import { Tool } from '@langchain/core/tools';

/**
 * Construct and return a *bare* LangChain AgentExecutor.
 * 
 * This creates an agent that can use tools but does not include memory.
 * Memory is added separately via RunnableWithMessageHistory.
 */
export function createAgent(
  llm: BaseChatModel,
  tools: Tool[],
  prompt: ChatPromptTemplate
): AgentExecutor {
  const agent = createToolCallingAgent({
    llm,
    tools,
    prompt,
  });

  return new AgentExecutor({
    agent,
    tools,
    verbose: true,
  });
} 